salt/_modules/caasp_etcd.py

```diff
@@ -1,14 +1,37 @@
 from __future__ import absolute_import
+
 import logging
 
+log = logging.getLogger(__name__)
 
-LOG = logging.getLogger(__name__)
-DESIRED_MEMBER_COUNT = 3
+# minimum number of etcd masters we recommend
+RECOMMENDED_MEMBER_COUNT = 3
 
 
 def __virtual__():
     return "caasp_etcd"
 
+
+# Grain used for getting nodes
+# make sure 'network.interfaces' is in the pillar/mine.sls
+_GRAIN = 'network.interfaces'
+
+
+def _get_num_kube(expr):
+    """
+    Get the number of kubernetes nodes that in the cluster that match "expr"
+    """
+    log.debug("Finding nodes that match '%s' in the cluster", expr)
+    nodes = __salt__['mine.get'](expr, _GRAIN, expr_form='grain').values()
+    # 'mine.get' is not available in the master, so it will return nothing
+    # in that case, we can try again with saltutil.runner... uh?
+    if not nodes:
```

Comment:
This is really nasty. SaltSatck does not allow to invoke `mine.get` in the master _unless_ we use the `saltutil.runner`...:-(

Reply:
I would move that comment into the source code :)

Reply:
I think the comment in lines 37-38 basically say that...